operations/security_and_authentication/Change_NCN_Image_Root_Password_and_SSH_Keys.md

```diff
@@ -229,162 +246,162 @@ The Ceph image Ceph image is used by the utility storage nodes.
+   ncn-mw# cd ../..
    ```
 
 1. The Ceph image now has the image changes.
 
 1. Update BSS with the new image for utility storage nodes.
 
-   **WARNING:** If doing a CSM software upgrade, skip this section to continue with Common Cleanup.
+   **WARNING:** If doing a CSM software upgrade, then skip this section and proceed to [Common cleanup](#common-cleanup).
 
    > If not doing a CSM software upgrade, this process will update the entries in BSS for the utility storage nodes to use the new Ceph image.
    >
    > 1. Set all utility storage nodes to use newly created Ceph image.
    >
-   >     This will use the CEPHVERSION and CEPHNEW variables defined earlier.
+   >     This will use the `CEPHVERSION` and `CEPHNEW` variables defined earlier.
    >
    >     ```bash
-   >     ncn-m# for node in $(grep -oP "(ncn-s\w+)" /etc/hosts | sort -u)
-   >     do
-   >       echo $node
-   >       xname=$(ssh $node cat /etc/cray/xname)
-   >       echo $xname
-   >       cray bss bootparameters list --name $xname --format json > bss_$xname.json
-   >       sed -i.old "s@ceph/${CEPHVERSION}@ceph/${CEPHNEW}@g" bss_$xname.json
-   >       kernel=$(cat bss_$xname.json | jq '.[]  .kernel')
-   >       initrd=$(cat bss_$xname.json | jq '.[]  .initrd')
-   >       params=$(cat bss_$xname.json | jq '.[]  .params')
-   >       cray bss bootparameters update --initrd $initrd --kernel $kernel --params $params --name $xname --format json
-   >     done
+   >     ncn-mw# for node in $(grep -oP "(ncn-s\w+)" /etc/hosts | sort -u)
+   >             do
+   >                 echo $node
+   >                 xname=$(ssh $node cat /etc/cray/xname)
+   >                 echo $xname
+   >                 cray bss bootparameters list --name $xname --format json > bss_$xname.json
+   >                 sed -i.$(date +%Y%m%d_%H%M%S%N).orig "s@/ceph/${CEPHVERSION}\([\"/[:space:]]\)@/ceph/${CEPHNEW}\1@g" bss_$xname.json
+   >                 kernel=$(cat bss_$xname.json | jq '.[]  .kernel')
+   >                 initrd=$(cat bss_$xname.json | jq '.[]  .initrd')
+   >                 params=$(cat bss_$xname.json | jq '.[]  .params')
+   >                 cray bss bootparameters update --initrd $initrd --kernel $kernel --params "$params" --hosts $xname --format json
+   >             done
    >     ```
 
-## Common Cleanup
+## Common cleanup
 
 1. Remove the workarea so the space can be reused.
 
    ```bash
-   ncn-m# rm -rf /run/initramfs/overlayfs/workingarea
+   ncn-mw# rm -rf /run/initramfs/overlayfs/workingarea
    ```
 
-## Deploy Changes
+## Deploy changes
 
 1. Rebuild nodes.
 
-   **WARNING:** If doing a CSM software upgrade, skip this step since the upgrade process does a rolling rebuild with some additional steps.
+   **WARNING:** If doing a CSM software upgrade, then skip this step since the upgrade process does a rolling rebuild with some additional steps.
```

Comment:
```suggestion
   **WARNING:** If doing a CSM software upgrade, then skip this step because the upgrade process does a rolling rebuild with some additional steps.
```